heat/fft/fft.py

```diff
@@ -0,0 +1,787 @@
+    n : int, optional
+        Length of the transformed axis of the output. If not given, the length is taken to be the length of the input
+        along the axis specified by axis. If `n` is smaller than the length of the input, the input is cropped. If `n` is
+        larger, the input is padded with zeros. Default: None.
+    axis : int, optional
+        Axis over which to compute the FFT. If not given, the last axis is used, or the only axis if x has only one
+        dimension. Default: -1.
+    norm : str, optional
+        Normalization mode: 'forward', 'backward', or 'ortho' (see `numpy.fft` for details). Default is "backward".
+
+    Notes
+    -----
+    This function requires MPI communication if the input array is transformed along the distribution axis.
+    If the input array is 1-D and distributed, this function copies the entire array on each MPI process! i.e. if the array is very large, you might run out of memory.
+    Hint: if you are looping through a batch of 1-D arrays to transform them, consider stacking them into a 2-D DNDarray and transforming them all at once (see :func:`fft2`).
+    """
+    return __fft_op(x, torch.fft.fft, n=n, axis=axis, norm=norm)
+
+
+def fft2(
+    x: DNDarray, s: Tuple[int, int] = None, axes: Tuple[int, int] = (-2, -1), norm: str = None
+) -> DNDarray:
+    """
+    Compute the 2-dimensional discrete Fourier Transform.
+
+    This function computes the 2-dimensional discrete Fourier Transform over the specified axes in an M-dimensional
+    array by means of the Fast Fourier Transform (FFT). By default, the last two axes are transformed, while the
+    remaining axes are left unchanged.
+
+    Parameters
+    ----------
+    x : DNDarray
+        Input array, can be complex
+    s : Tuple[int, int], optional
+        Shape of the output along the transformed axes. (default is x.shape)
+    axes : Tuple[int, int], optional
+        Axes over which to compute the FFT. If not given, the last `len(s)` axes are used, or all axes if `s` is also
+        not specified. Repeated indices in `axes` means that the transform over that axis is performed multiple times.
+        (default is (-2, -1))
+    norm : str, optional
+        Normalization mode: 'forward', 'backward', or 'ortho' (see `numpy.fft` for details). Default is "backward".
+
+    Notes
+    -----
+    This function requires MPI communication if the input array is distributed and the split axis is transformed.
+    """
+    return __fftn_op(x, torch.fft.fft2, s=s, axes=axes, norm=norm)
+
+
+def fftn(
+    x: DNDarray, s: Tuple[int, ...] = None, axes: Tuple[int, ...] = None, norm: str = None
+) -> DNDarray:
+    """
+    Compute the N-dimensional discrete Fourier Transform.
+
+    This function computes the N-dimensional discrete Fourier Transform over any number of axes in an M-dimensional
+    array by means of the Fast Fourier Transform (FFT). By default, all axes are transformed, with the real transform
```

Comment:
The comment on real transform along one axis and complex transform on the others does not appear in PyTorch. Is there a difference of the conventions of PyTorch and NumPy?

Reply:
Interesting, this is GitHub copilot extrapolating out of context. The comment on the real transform belongs in `rfftn`. I've modified the docs.